rsa.go

```diff
@@ -15,29 +15,17 @@ type SigningMethodRSA struct {
 
 // Specific instances for RS256 and company
 var (
-	SigningMethodRS256 *SigningMethodRSA
-	SigningMethodRS384 *SigningMethodRSA
-	SigningMethodRS512 *SigningMethodRSA
+	SigningMethodRS256 = NewSigningMethodRSA("RS256", crypto.SHA256)
+	SigningMethodRS384 = NewSigningMethodRSA("RS384", crypto.SHA384)
+	SigningMethodRS512 = NewSigningMethodRSA("RS512", crypto.SHA512)
 )
 
-func init() {
-	// RS256
-	SigningMethodRS256 = &SigningMethodRSA{"RS256", crypto.SHA256}
-	RegisterSigningMethod(SigningMethodRS256.Alg(), func() SigningMethod {
-		return SigningMethodRS256
-	})
-
-	// RS384
-	SigningMethodRS384 = &SigningMethodRSA{"RS384", crypto.SHA384}
-	RegisterSigningMethod(SigningMethodRS384.Alg(), func() SigningMethod {
-		return SigningMethodRS384
-	})
-
-	// RS512
-	SigningMethodRS512 = &SigningMethodRSA{"RS512", crypto.SHA512}
-	RegisterSigningMethod(SigningMethodRS512.Alg(), func() SigningMethod {
-		return SigningMethodRS512
-	})
+// NewSigningMethodRSA creates a new SigningMethodRSA struct and
+// registers it as a signing method.
+func NewSigningMethodRSA(name string, hash crypto.Hash) *SigningMethodRSA {
```

Comment:
I can change the functions so that they start with `newSigningMethod` instead.

Reply:
I updated the pull request. :heavy_check_mark:

Reply:
LIked this, was a good afterthought. Now I think it should be ready to go, it LGTM from my side

Reply:
@lggomez GitHub believes there is one outstanding request for change.

Could you please take a look if this is still the case, because I can't spot it?